elastic/kibana
# [Fleet] Add option to have Kafka dynamic topics in outputs

## Summary

Add option to have Kafka dynamic topics in outputs settings. It adds selection with radio buttons under the `topic` section.

### UI changes

![Screenshot 2024-09-13 at 16 32 54](https://github.com/user-attachments/assets/9414f475-c115-4986-a204-eeb79471b7ea)
![Screenshot 2024-09-13 at 16 33 03](https://github.com/user-attachments/assets/ac36b1da-0e81-4f97-9632-ced8b5776af2)
![Screenshot 2024-09-13 at 16 33 12](https://github.com/user-attachments/assets/5ea3cf66-d929-45fb-aa3b-924734c34305)
![Screenshot 2024-09-13 at 16 33 40](https://github.com/user-attachments/assets/62e477c8-5719-41fe-94b1-4bdda0f00007)

<details>
  <summary>Testing</summary>

- Create a kafka output, fill the required fields, select "static" topic and fill it out with "default_topic". Payload will be:
![Screenshot 2024-09-13 at 16 36 50](https://github.com/user-attachments/assets/ea3761b8-39cb-40ce-bf1c-85e3af1f9d4b)
- The output should be created correctly

- Create a kafka output, fill the required fields, select "dynamic" topic and choose one from the dropdown. Payload will be:
![Screenshot 2024-09-13 at 16 36 23](https://github.com/user-attachments/assets/8d0d3ce4-5fb0-4c01-a939-b6a9d50da219)

- Create a kafka output, fill the required fields, select "dynamic" topic and input one not present in the dropdown (I had "custom_topic". Payload will be like this:
![Screenshot 2024-09-13 at 16 35 52](https://github.com/user-attachments/assets/27347d01-3efc-4c23-85a8-f3287854de4f)
</details>

- Try updating an existing Kafka output from one type to another, it should work correctly

### Checklist
